python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -117,15 +117,28 @@ def registerDatasetType(self, datasetType):
 
         Raises
         ------
-        KeyError
-            Dataset is already registered.
         ValueError
-            DatasetType is not valid for this registry.
+            DatasetType is not valid for this registry or is already registered
+            but not identical.
         """
+        # Check if datasetType is valid for this Registry
         if not self._isValidDatasetType(datasetType):
             raise ValueError("DatasetType is not valid for this registry")
-        if datasetType.name in self._datasetTypes:
-            raise KeyError("DatasetType: {} already registered".format(datasetType.name))
+        # If a DatasetType is already registered it must be identical
+        try:
+            # A DatasetType entry with this name may exist, get it first.
+            # Note that we can't just look in the cache, because it may not be there yet.
+            existingDatasetType = self.getDatasetType(datasetType.name)
+        except KeyError:
+            # No registered DatasetType with this name exists, move on to inserting it
+            pass
+        else:
+            # A DatasetType with this name exists, check if is equal
+            if datasetType == existingDatasetType:
+                return
+            else:
+                raise ValueError("DatasetType: {} already registered but different".format(datasetType.name))
```

Comment:
I guarantee we'll want an error message that shows how they differ.  Might be a to-do item rather than something to do on this ticket.